tephra-hbase-compat-1.1-base/src/main/java/org/apache/tephra/hbase/coprocessor/TransactionProcessor.java

```diff
@@ -303,10 +305,22 @@ public InternalScanner preCompactScannerOpen(ObserverContext<RegionCoprocessorEn
     // Get the latest tx snapshot state for the compaction
     TransactionVisibilityState snapshot = cache.getLatestState();
 
-    // Record tx state before the compaction
-    if (compactionState != null) {
+    if (pruneEnable == null) {
+      Configuration conf = getConfiguration(c.getEnvironment());
+      pruneEnable = conf.getBoolean(TxConstants.TransactionPruning.PRUNE_ENABLE,
```

Comment:
Since this code does not change between Tehpra and derived class implementation, we may need to handle the case when configuration is not available after startup. What do you think?